### DISCUSSION TITLE
有考虑支持符号链接形式的路径监听吗

### DISCUSSION THREAD

**Discussion by Sorades** [shion-app/shion#61] [Ideas]
OS: Windows 11, version 23H2
shion version: v0.13.1

例如scoop管理的软件或WindowsApps每次更新后都会改变安装路径，提供链接到新版本的符号链接，shion目前并不支持对链接路径的监听，导致每次软件更新后需要手动修改监听路径

以scoop管理的qq-nt为例，其结构为：
qq-nt
├── 9.9.15_241009_01
├── 9.9.16_241023_01
└── current -> D:\Windows\Scoop\User\apps\qq-nt\9.9.16_241023_01
shion只能监听"D:\Windows\Scoop\User\apps\qq-nt\9.9.16_241023_01\Files\QQ.exe"，而无法监听"D:\Windows\Scoop\User\apps\qq-nt\current\Files\QQ.exe"

**Comment by hanaTsuk1** [comment]
我大致想了一下方案
1. 用户需要自行打开对应软件`符号链接`的设置开关和修改`路径`为符号链接路径
2. 然后软件才去找对应的源exe进行匹配

你觉得怎么样🤔

**Reply by Sorades** [reply]
直接设计成无感的会不会好一点，感觉没必要多加个开关🤔

**Reply by hanaTsuk1** [reply]
具我查到的信息，做不到的

只能通过链接后的路径（即current ）来查源exe，不能反过来

**Reply by Sorades** [reply]
Emmm 应该不用反过来查吧。按我的理解就是对用户指定的 `软件路径` 在建立监听之前先判断一下这是不是链接，如果是就解析为`源路径`，如果不是就不用变。
关于判断路径是不是符号链接，我有搜到：https://learn.microsoft.com/zh-cn/windows/win32/api/fileapi/nf-fileapi-getfileattributesw
其中文件属性`FILE_ATTRIBUTE_REPARSE_POINT`应该是指明该路径是否为链接

这样应该改的东西应该会少一些🙂

**Reply by hanaTsuk1** [reply]
> 先判断一下这是不是链接

判断这里，拿到的路径是“D:\Windows\Scoop\User\apps\qq-nt\9.9.16_241023_01\Files\QQ.exe”，如果拿到的是current，那根本没有这个问题
